### ISSUE TITLE
If you try to Run Quick Encounter on a QE without a Note, it creates a Note but does not immediately run the QE

### ISSUE THREAD

**Issue by spetzel2020** [spetzel2020/quick-encounters#83]
QE 1.0.1
Foundry 9.251

You are prompted to create a Note at the center of you View, and QE does create the Note. However you then have to press [Run Quick Encounter] (a second time) to actually run it.

**Comment by spetzel2020** [comment]
Also, in some cases you get the warning (in the JE) about there not being a Map Note and then it runs the QE anyway, placing the tokens in some arbitrary location?

**Comment by spetzel2020** [comment]
Fixed: EncounterNote.create() was creating an array of NoteDocuments in Foundry v8+; changed to pick element[0]
(although I don't think this is the real problem)

**Comment by spetzel2020** [comment]
I believe the problem arises because of a race condition in this code in QuickEncounter.run()
```
            let mapNote = qeJournalEntry.clickedNote;
            if (!mapNote) {
                // Switch to the correct scene if confirmed
                let qeScene = QuickEncounter.getEncounterScene(qeJournalEntry);
                //If there isn't a Map Note anywhere, prompt to create one in the center of the view
                if (!qeScene) {
                    EncounterNote.noMapNoteDialog(this);
                    //Try again now that it should have a scene if yo responded yes
                    qeScene = QuickEncounter.getEncounterScene(qeJournalEntry);
                }
                const isPlaced = await EncounterNote.mapNoteIsPlaced(qeScene, qeJournalEntry);
                if (!isPlaced ) {return;}
                //Something is desperately wrong if this is null
                mapNote = qeJournalEntry.sceneNote;
            }
            this.sourceNoteData = mapNote.data;
```
EncounterNote.noMapNoteDialog() creates an `async place()` operation which has not completed by the time the `qeScene = ...getEncounterScene()` runs

Possible solutions: Put this on a delay OR just add text to the dialog that explains you need to rerun the Quick Encounter after placing the Map Note.

**Comment by spetzel2020** [comment]
With QE 1.0.3, after running from a Note, it will run the QE without prompting, probably because `clickedNote` is still set.
Confirmed that after a Refresh (F5) this works correctly (prompting for the Note)

**Comment by spetzel2020** [comment]
Having tried other solutions to this, I am reverting to the one showing a dialog informing the user to "Check the Map Note placement and rerun the QE"

**Comment by spetzel2020** [comment]
Fixed in 1.1.5e (not yet released)

**Comment by spetzel2020** [comment]
Released in 1.1.5
